styled/styled.go

```diff
@@ -0,0 +1,140 @@
+package styled
+
+import (
+	"fmt"
+
+	"github.com/elves/elvish/eval/vals"
+)
+
+type TextStyle struct {
+	bold       *bool
```

Comment:
Is `nil` (lack of an attribute) and `false` (explicit false) a useful distinction? I think not - just make the fields `bool` and the code can be simplified quite a bit.

Reply:
You are right. `no-` is probably a better prefix.